xyz-psql-connector/src/main/java/com/here/xyz/psql/PSQLXyzConnector.java

```diff
@@ -129,38 +129,73 @@ protected XyzResponse processGetFeaturesByBBoxEvent(GetFeaturesByBBoxEvent event
+        }
+        else
+        { tweakType   = TweaksSQL.ENSURE;
+          event.setTweakType( TweaksSQL.ENSURE );
+          tweakParams = new HashMap<String, Object>();
+          switch( event.getVizSampling().toLowerCase() )
+          { case "high" : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 10 ) ); break;
+            case "low"  : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 70 ) ); break;    
+            case "off"  : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 100 )); 
+                          bVizSamplingOff = true;
+                          break;    
+            case "med"  : 
+            default     : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 20 ) ); break;    
+          }
+        }
+
+        int distStrength = 0;
+
+        switch (tweakType) {
 
           case TweaksSQL.ENSURE: {
             int rCount = executeQueryWithRetry(SQLQueryBuilder.buildEstimateSamplingStrengthQuery(event, bbox )).getFeatures().get(0).get("rcount");
 
             boolean bDefaultSelectionHandling = (tweakParams.get(TweaksSQL.ENSURE_DEFAULT_SELECTION) == Boolean.TRUE );
 
-            if( event.getSelection() == null && !bDefaultSelectionHandling )
+            if( event.getSelection() != null && "*".equals( event.getSelection().get(0) ) )
+             event.setSelection(null);
+            else if( event.getSelection() == null && !bDefaultSelectionHandling )
              event.setSelection(Arrays.asList("id","type"));
+                        
+            distStrength = TweaksSQL.calculateDistributionStrength( rCount, Math.max(Math.min((int) tweakParams.getOrDefault(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD,10),100),10) * 1000 );
 
-            int chunkSize    = Math.max(Math.min((int) tweakParams.getOrDefault(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD,10),100),10) * 1000,
-                distStrength = TweaksSQL.calculateDistributionStrength( rCount, chunkSize );
-
-            if( distStrength == 0 ) break; // NrOfObjects less than chunkSize -> fall back to non-tweaks usage
-            HashMap<String, Object> hmap = new HashMap<String, Object>();
+            HashMap<String, Object> hmap = new HashMap<String, Object>();    
             hmap.put("algorithm", new String("distribution"));
             hmap.put("strength", new Integer( distStrength ));
             tweakParams = hmap;
             // fall thru tweaks=sampling
           }
 
           case TweaksSQL.SAMPLING: {
-            FeatureCollection collection = executeQueryWithRetry(SQLQueryBuilder.buildSamplingTweaksQuery(event, bbox, tweakParams, dataSource));
-            collection.setPartial(true);
-            return collection;
-          }
-
-          case TweaksSQL.SIMPLIFICATION: {
-            FeatureCollection collection = executeQueryWithRetry(SQLQueryBuilder.buildSimplificationTweaksQuery(event, bbox, tweakParams, dataSource));
+            if( bTweaks || !bVizSamplingOff )
```

Comment:
bTweaks is aways true

Reply:
false assumption. bTweaks is used to distingush whether "tweaks" (bTweaks = true) pararmeter or "mode" (bOptViz = true) has been specified on url.